core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -231,11 +231,15 @@ class KafkaApis(val requestChannel: RequestChannel,
       val responseBody = new OffsetCommitResponse(results.asJava)
       requestChannel.sendResponse(new RequestChannel.Response(request, new ResponseSend(request.connectionId, responseHeader, responseBody)))
     } else {
-      // filter non-existent topics
-      val invalidRequestsInfo = offsetCommitRequest.offsetData.asScala.filter { case (topicPartition, _) =>
-        !metadataCache.contains(topicPartition.topic)
+
+      val (nonExistingRequestsInfo, existingTopicsRequestsInfo) = offsetCommitRequest.offsetData.asScala.toMap.partition {
+        case (topicPartition, _) => !metadataCache.contains(topicPartition.topic)
+      }
+
+      // treat topics with Describe ACL as if they were non-existent
+      val (filteredRequestInfo, unauthorizedForDescribeRequestInfo) = existingTopicsRequestsInfo.partition {
```

Comment:
Seems like it would be possible to merge these first two partition calls. Something like this:
```scala
val (existingTopicInfo, nonExistingTopicInfo) = offsetCommitRequest.offsetData.asScala.toMap.partition {
  case (topicPartition, _) => authorize(request.session, Describe, new Resource(Topic, topicPartition.topic)) && 
     metadataCache.contains(topicPartition.topic)
}
```
That might also make the expectation that you need Describe access to check topic existence clearer. Thoughts?